File: SharedLibrary/Shared/Objects/WebStatus.cs
using System;
using System.Collections.Generic;
using System.Text;
using JetBrains.Annotations;
using SharedLibrary.Enums;

namespace SharedLibrary
{
	[Android.Runtime.Preserve(AllMembers = true)]
	public class WebStatus
	{
		public PageStatus Status { get; set; }

		public string Message { get; set; }

		public object Content { get; set; }

		public bool CanAccess => (Status == PageStatus.Functional);

		public WebStatus() { }

		public WebStatus(PageStatus status, string msg = null, object content = null)
		{
			this.Status = status;
			this.Message = msg;
			this.Content = content;
		}
	}
}

File: MobileApp/zoliky_teachers/lib/utils/api/models/WebStatus.dart
import 'package:zoliky_teachers/utils/api/enums/PageStatus.dart';

class WebStatus {
  PageStatus status;
  String message;
  Object content;
  bool get canAccess => (status == PageStatus.Functional);

  WebStatus();

  WebStatus.fromJson(Map<String, dynamic> json)
      : status = PageStatus(json['Status']),
        message = json['Message'],
        content = json['Content'];

  WebStatus.unFunctional({Object content})
      : status = PageStatus.Unfunctional,
        message = "Nezdařilo se načíst stav serverů",
        content = content ?? null;

  @override
  String toString() {
    return "Status: $status, \nmsg: $message, \nContent: $content\n";
  }
}
